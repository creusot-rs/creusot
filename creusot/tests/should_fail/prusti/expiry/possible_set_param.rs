extern crate creusot_contracts;
use creusot_contracts::prusti_prelude::*;

#[logic('x)]
#[after_expiry('a, result == *x)]
fn test<'a>(x: &'a mut u32) -> u32 {
    0u32
}