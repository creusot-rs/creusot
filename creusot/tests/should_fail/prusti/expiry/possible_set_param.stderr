error: Cannot move to the state set 'a since it might have been instantiated with multiple states
 --> possible_set_param.rs:5:16
  |
5 | #[after_expiry('a, result == *x)]
  |                ^^

error: aborting due to 1 previous error

